main.cpp

```diff
@@ -0,0 +1,87 @@
+#include <stdio.h>
+#include <string.h>
+#include <iostream>
+#include <fstream>
+// #include <memory.h>
+
+
+// we need to reference this for a better sequence data
+// typedef struct
+// {
+//     FILE *fh;
+//     int width, height;
+//     int next_frame;
+//     int seq_header_len, frame_header_len;
+//     int frame_size;
+// } depth_input_t;
+
+#define DEPTH_MAGIC "D16"
+#define MAX_DEPTH_HEADER 80
+#define Y4M_FRAME_MAGIC "FRAME"
+#define MAX_FRAME_HEADER 80
+
+int height;
+int width;
+typedef struct depth_meta
+{
+    int undefined;
+    int undefined;
+    int undefined; /* data */
+} depth_meta;
+depth_meta metadata;
```

Comment:
We need to think of how to define the meta data, should it be linked with the depth precision?